### PR TITLE
fix: web-channel handling grpc and http errors

### PR DESCRIPTION
**Pull request by thanhdanh**
## Summary
We sometimes catch the error `(BUG) unexpectedly received no response` that mentions an empty response.  It could be the HTTP errors when connecting to some node proxies or GRPC web (or timeout) errors that we are not handling now. We don't catch them now before decoding the response, so we cannot retry it. If we get this `no response` issue in the wallet, the wallet will be stuck there.

This PR improves:
- Add retry logics on new grpc errors and HTTP errors.
- Check `grpc-status` and `grpc-message` response headers before decoding response body.
- Catch http-errors

### CODE REVIEW

**Review by feri42 — APPROVED** [review]
utACK

File: src/http/HttpError.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import HttpStatus from "./HttpStatus";
```
Comment (ChALkeR) [inline src/http/HttpError.js:1]: Should have used a full path, otherwise breaks on Node.js
Fix in 17ab90de0b507ff0c49c0a499523e5f95edb900a

Comment (feri42) [reply src/http/HttpError.js:1]: cc @thanhdanh Has this been addressed?

Comment (thanhdanh) [reply src/http/HttpError.js:1]: It's fixed in the above commit and published

**Pull request merged**